### PR TITLE
Add Azure cluster-autoscaler e2e test

### PR DESCRIPTION
**Pull request by nojnhuh**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

This PR adds the first real functional test of the Azure provider for cluster-autoscaler. The test creates a bunch of Pods, waits for the cluster to scale up, deletes the pods, and waits for the cluster to scale back down.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/cloudprovider/azure/test/Makefile [hunk 1]
```diff
@@ -46,8 +46,15 @@ install-e2e: $(HELM)
 		--set image.repository=$(IMAGE)-$(GOARCH) \
 		--set image.tag=$(TAG) \
 		--set image.pullPolicy=Always \
+		--set extraArgs.scale-down-delay-after-add=10s \
```
Comment (nojnhuh) [inline cluster-autoscaler/cloudprovider/azure/test/Makefile:49]: Basically what I was aiming for with this new config was "react as fast as possible" so any feedback on how to best achieve that is most welcome.

File: cluster-autoscaler/cloudprovider/azure/test/e2e/azure_test.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Azure Provider", func() {
+	var (
+		namespace *corev1.Namespace
+	)
+
+	BeforeEach(func() {
+		Eventually(allVMSSStable, "10m", "30s").Should(Succeed())
```
Comment (nojnhuh) [inline cluster-autoscaler/cloudprovider/azure/test/e2e/azure_test.go:21]: All of these timeouts are kind of arbitrary, so I'm totally open to tweaking these.

File: cluster-autoscaler/cloudprovider/azure/test/templates/cluster-template-prow-aks-aso-cluster-autoscaler.yaml [hunk 3]
```diff
@@ -162,11 +162,6 @@ spec:
       mode: System
       owner:
         name: ${CLUSTER_NAME}
-      tags:
```
Comment (nojnhuh) [inline cluster-autoscaler/cloudprovider/azure/test/templates/cluster-template-prow-aks-aso-cluster-autoscaler.yaml:165]: Should we try to isolate things like kube-system Pods from the side effects of these tests? That's what I was trying to achieve here by removing the autoscaler tags from this System pool but I can still see things like coredns sometimes scheduling onto User pools.

File: cluster-autoscaler/cloudprovider/azure/test/Makefile (hunk 1 above)
Comment (jackfrancis) [reply cluster-autoscaler/cloudprovider/azure/test/Makefile:49]: sgtm

File: cluster-autoscaler/cloudprovider/azure/test/templates/cluster-template-prow-aks-aso-cluster-autoscaler.yaml (hunk 3 above)
Comment (jackfrancis) [reply cluster-autoscaler/cloudprovider/azure/test/templates/cluster-template-prow-aks-aso-cluster-autoscaler.yaml:165]: I'm not sure about the connection here, but I think at this phase of the game this is good enough, and we're in good shape to raply make tweaks to cover these types of things that make testing cluster-autoscaler tricky.

**Comment by jackfrancis** [comment]
this is a great start!

/lgtm
/approve

**Comment by willie-yao** [comment]
Sorry I was late to this, lgtm!!

**Pull request merged**